webserver/netty/src/main/java/io/helidon/webserver/netty/BareResponseImpl.java

```diff
@@ -117,6 +123,11 @@ public void writeStatusAndHeaders(Http.ResponseStatus status, Map<String, List<S
             response.headers().add(headerEntry.getKey(), headerEntry.getValue());
         }
 
+        // Copy HTTP/2 headers to response for correlation (streamId)
```

Comment:
Unrelated to this change... Are there other headers that should also be copied for the same reason? I have no specific ones to  nominate, I'm just wondering.

Reply:
I wanted to emphasized the experimental nature of the feature to make sure people use this at their own risk. Adding "experimental" as a prefix/suffix on every property was awkward.

Reply:
As for the header copying part, I can't think of a reason to do that for HTTP 1.X. The need for that is very specific to the way HTTP/2 support is implemented.

Reply:
Re: `experimental`... The implication would be that the config for _any_ experimental feature would be under `experimental`. When any such feature changes from experimental to fully supported then not just the key but also the structure of the config changes. To minimize changes to config files during that transition what about
```
server:
    http2-experimental:
        enable:
        max-content-length:
```
When the feature matures and `http2-experimental` changes to `http2` then one line in the config file changes and the code could be written to require one change to a constant, but the basic structure remains the same which helps minimize coding changes. This would also help segregate the config settings from potentially different experimental features from each other.

That also follows the principle in your note that adding `experimental` to each property name (which I didn't mean to suggest) would be awkward. Is adding `http2` to each property name in the original proposal equally awkward?

Reply:
Your proposal address my concern: experimental features should be clearly marked. However, I doubt any of these properties will survive the transition from experimental to production ready. First, we should probably have HTTP/2 enabled all the time (like browsers today that automatically pick the best version) and, second, the max length is an artificial byproduct of the current implementation and will hopefully go away too.

I have a slight preference to have a single `experimental` section for new features like these rather than `feature1-experimental`, `feature2-experimental`, etc. since the latter would require changes to `ServerConfiguration.java` every time. Using one `experimental` section provides a single extension point for all these type of features. Alternatively we could have:

```
server:
  experimental:
    http2:
      enable:
      max-content-length:
```

This alternative would be fine by me too. But as I said above, I doubt the `http2` section will exist as such when this feature is no longer experimental, so that shouldn't be a reason to opt for one.

Reply:
I think it makes sense having an `http2` section as in your latest example, whether the config values are  experimental-only or not.

I'm not seeing why introducing `experimental` as its own level (vs. the `feature-1-experimental` approach) saves us from edits to `ServerConfiguration.java` as features mature from experimental to production. But in any event I'm fine with your latest proposed structure. In general if the `feature-1` section remained when the feature matured, promoting it to be under `server` instead of `experimental` would involve only a single tiny code change.

Reply:
> I'm not seeing why introducing `experimental` as its own level (vs. the `feature-1-experimental` approach) saves us from edits to `ServerConfiguration.java` as features mature from experimental to production. 

Suppose, as is likely going to be in for HTTP/2, that these properties go away when moving to production. In that case, we can leave the `experimental` section empty (or in use by some other experimental feature) without touching `ServerConfiguration.java`. If we have an `http2-experimental` section, removing it (as it would no longer make sense to have) after production would require changes to `ServerConfiguration.java`.